infra/experimental/sanitizers/ExecSan/PoEs/pytorch-lightning-1.5.10/fuzz_pytorch_lightning.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/local/bin/python3
+
+# Copyright 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+"""Exploit pytorch lightning with fuzzer's input as a random env variable.
+
+This PoC is extended from a report in GitHub Advisory Database:
+https://github.com/advisories/GHSA-r5qj-cvf9-p85h
+The original report documents an exploit using a specific environment variable,
+we show a way to achieve the same exploit with an arbitrary env variable.
+"""
+
+import os
+import random
+import string
+import sys
+import atheris
+
+with atheris.instrument_imports():
+  from pytorch_lightning import Trainer
+  from pytorch_lightning.utilities.argparse import parse_env_variables
+
+
+def random_env_name(length):
+  """Generate a valid random name of environment variables"""
+  valid_env_head_chars = string.ascii_uppercase + string.ascii_lowercase + '_'
+  valid_env_tail_chars = valid_env_head_chars + string.digits
+
+  env_head = random.choice(valid_env_head_chars)
+  env_tail = ''.join(random.choice(valid_env_tail_chars) \
+                     for _ in range(random.randint(0, length)))
+  return env_head + env_tail
+
+
+def prepare_fuzzing_input(data):
+  """Prepare the data needed by the exploit with input data from fuzzers."""
+  data = data.replace(b'\0', b'')
+  env_name = random_env_name(len(data))
+  return data, env_name
+
+
+def exploit_target(env_value, env_name):
+  """This target is based on a snippet from
```

Comment:
this is formatted weirdly. Can your editor change this so that each line is as close to 80 chars as possible.

Reply:
By the way, what is the conventional way to fit URLs into a line of 80 chars?
E.g. [this line](https://github.com/google/oss-fuzz/blob/36aa9c9ce108dd324ae121ef33262c461fc50e09/infra/experimental/sanitizers/ExecSan/PoEs/pytorch-lightning-1.5.10/fuzz_pytorch_lightning.py#L58).

Reply:
I think it's OK if URLs make a line too long.

Reply:
yeah just add a `pylint: disable=line-length` (not sure if I got that exactly right)

Reply:
It might not even be necessary I think pylint might be smart enough to recognize when there's a URL.

Reply:
Yep, you right:`pylint` did not complain.
I was just curious how people normally do it : )